packages/@aws-cdk/aws-appmesh/lib/gateway-route-spec.ts

```diff
@@ -1,32 +1,408 @@
+  public static hostname(hostname: GatewayRouteHostname, options?: GrpcGatewayRouteHostnameMatchOptions): GrpcGatewayRouteMatch {
+    return new GrpcGatewayRouteHostnameMatchImpl(hostname, options);
+  }
+
+  /**
+   * Create metadata based gRPC gateway route match.
+   *
+   * @param metadata An object that represents the data to match from the request.
+   * @param options Optional properties to define gRPC gateway route match
+   */
+  public static metadata(metadata: GrpcMetadataMatch[], options?: GrpcGatewayRouteMetadataMatchOptions): GrpcGatewayRouteMatch {
+    return new GrpcGatewayRouteMetadataMatchImpl(metadata, options);
+  }
+
+  /**
+   * Create service name based gRPC gateway route match.
+   *
+   * @param serviceName The fully qualified domain name for the service to match from the request
+   * @param options Optional properties to define gRPC gateway route match
+   */
+  public static serviceName(serviceName: string, options?: GrpcGatewayRouteServiceNameMatchOptions): GrpcGatewayRouteMatch {
+    return new GrpcGatewayRouteServicenameMatchImpl(serviceName, options);
+  }
+
+  /**
+   * Return the gRPC gateway route match configuration.
+   */
+  abstract bind(scope: Construct): GrpcGatewayRouteMatchConfig;
+}
+
+class GrpcGatewayRouteHostnameMatchImpl extends GrpcGatewayRouteMatch {
+  constructor(
+    private readonly hostname: GatewayRouteHostname,
+    private readonly options?: GrpcGatewayRouteHostnameMatchOptions,
+  ) { super(); }
+
+  bind(scope: Construct): GrpcGatewayRouteMatchConfig {
+    return {
+      requestMatch: {
+        hostname: this.hostname.bind(scope).hostnameMatch,
+        metadata: this.options?.metadata?.map(metadata => metadata.bind(scope).metadataMatch),
+        serviceName: this.options?.serviceName,
+      },
+    };
+  }
+}
+
+class GrpcGatewayRouteMetadataMatchImpl extends GrpcGatewayRouteMatch {
+  constructor(
+    private readonly metadata: GrpcMetadataMatch[],
+    private readonly options?: GrpcGatewayRouteMetadataMatchOptions,
+
+  ) { super(); }
+
+  bind(scope: Construct): GrpcGatewayRouteMatchConfig {
+    return {
+      requestMatch: {
+        metadata: this.metadata.map(metadata => metadata.bind(scope).metadataMatch),
+        hostname: this.options?.hostname?.bind(scope).hostnameMatch,
+        serviceName: this.options?.serviceName,
+      },
+    };
+  }
+}
+
+class GrpcGatewayRouteServicenameMatchImpl extends GrpcGatewayRouteMatch {
+  constructor(
+    private readonly serviceName: string,
+    private readonly options?: GrpcGatewayRouteServiceNameMatchOptions,
+
+  ) { super(); }
+
+  bind(scope: Construct): GrpcGatewayRouteMatchConfig {
+    return {
+      requestMatch: {
+        serviceName: this.serviceName,
+        hostname: this.options?.hostname?.bind(scope).hostnameMatch,
+        metadata: this.options?.metadata?.map(metadata => metadata.bind(scope).metadataMatch),
+      },
+    };
+  }
+}
+
+/**
+ * Enum of rewrite default target
+ */
+export enum Default {
```

Comment:
The name `Default` is too generic. We can make this named something like `DefaultTarget`

Reply:
Addressed